### ISSUE TITLE
Add NC Account with "User Account creation" policy disabled

### ISSUE THREAD

**Issue by dloxterm** [nextcloud/android#10212]
### Is your feature request related to a problem? Please describe.
We're running Android with a mobile device management. Account creation is disabled since we don't want our users to create Google Accounts etc.
When opening the Nextcloud App I'll simply get "Forbidden by Admin" (the Device Administrator Dialog on Android) and the App will crash.
Android 11 with NC 3.20.1

### Describe the solution you'd like
We would like the NC app to behave like Davx5 for example.I'm not aware of the technical background, but for some reason I can add DAV Accounts  in Davx5 with the "Disable adding new users and profiles" policy enabled. 
Both Apps are pre-installed by default, so there should not be a difference in the settings...

### Describe alternatives you've considered
Allowing Account creation by the user. This is not viable for us. E.g. the Google Calendar App will ask for an Google Account.

### Additional context
Screenshot of the error
![image](https://user-images.githubusercontent.com/75365794/167360082-0152e38e-f6c3-4086-a7e8-a48f062a3118.png)

**Comment by AlvaroBrey** [comment]
> I'm not aware of the technical background, but for some reason I can add DAV Accounts in Davx5 with the "Disable adding new users and profiles" policy enabled

We use the Android system `AccountManager` to store our accounts. `DavX5` probably just stores them in its own database.

**Comment by dloxterm** [comment]
> We use the Android system AccountManager to store our accounts. DavX5 probably just stores them in its own database.

I would agree with you as other apps do it like this, but I can see that DAVx5 is adding the accounts here as well, so I'm not sure whether DAVx5 does not use `AccountManager`:

![image2](https://user-images.githubusercontent.com/75365794/167369421-73c60c0b-096e-4ef9-8717-d11e5be22ca9.png)

**Comment by joshtrichards** [comment]
Any chance you can capture the stack trace/crash output?

You may need to get the debug log: https://github.com/nextcloud/android#getting-debug-info-via-logcat-mag

Though it may be easier to simply install a recent Dev image via F-Droid (or download) then you can access the logs in-app via *Settings->Logs*.

**Comment by joshtrichards** [comment]
> We're running Android with a mobile device management. Account creation is disabled since we don't want our users to create Google Accounts etc.
When opening the Nextcloud App I'll simply get "Forbidden by Admin" (the Device Administrator Dialog on Android) and the App will crash.

Are you certain the failure is to due the account creation being disabled? I ask because the screenshot suggests the client was already set-up with an account...
